src/main/scala/viper/gobra/translator/encodings/arrays/ExclusiveArrayComponentImpl.scala

```diff
@@ -0,0 +1,64 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.encodings.arrays
+
+import viper.gobra.translator.encodings.EmbeddingComponent
+import viper.gobra.translator.interfaces.{Collector, Context}
+import viper.gobra.ast.{internal => in}
+import viper.silver.{ast => vpr}
+import ArrayEncoding.ComponentParameter
+import viper.gobra.translator.encodings
+
+class ExclusiveArrayComponentImpl extends ExclusiveArrayComponent {
+
+  override def finalize(col: Collector): Unit = {
+    emb.finalize(col)
+  }
+
+  /** Embeds Sequences of fixed length as specified by ComponentParameter. */
+  private val emb: EmbeddingComponent[ComponentParameter] = new encodings.EmbeddingComponent.Impl[ComponentParameter](
+    p = (e: vpr.Exp, id: ComponentParameter) => (_: Context) => vpr.EqCmp(vpr.SeqLength(e)(), vpr.IntLit(id._1)())(),
+    t = (id: ComponentParameter) => (_: Context) => vpr.SeqType(id._2)
+  )
+
+  /** Returns type of exclusive-array domain. */
+  override def typ(t: ComponentParameter)(ctx: Context): vpr.Type = emb.typ(t)(ctx)
+
+  /** Constructor of shared-array domain. */
```

Comment:
shouldn't this comment talk about exclusive arrays

Reply:
I changed it (not pushed yet).